save-backend/src/main/kotlin/com/saveourtool/save/backend/service/UserDetailsService.kt

```diff
@@ -78,29 +78,54 @@ class UserDetailsService(
      * @return UserSaveStatus
      */
     @Transactional
-    fun saveUser(user: User, oldName: String?): UserSaveStatus {
-        val userName = user.name
-        return if (oldName == null) {
-            userRepository.save(user)
-            UserSaveStatus.UPDATE
-        } else if (userRepository.validateName(userName) != 0L) {
-            userRepository.deleteHighLevelName(oldName)
-            userRepository.saveHighLevelName(userName)
-            userRepository.save(user)
-            UserSaveStatus.UPDATE
-        } else {
-            UserSaveStatus.CONFLICT
+    fun saveUser(user: User, oldName: String?): UserSaveStatus = if (oldName == null) {
+        userRepository.save(user)
+        UserSaveStatus.UPDATE
+    } else if (userRepository.validateName(user.name) != 0L) {
+        userRepository.deleteHighLevelName(oldName)
+        userRepository.saveHighLevelName(user.name)
+        userRepository.save(user)
+        UserSaveStatus.UPDATE
+    } else {
+        UserSaveStatus.CONFLICT
+    }
+
+    /**
+     * @param userNameCandidate
+     * @param userRole
+     * @return created [User]
+     */
+    @Transactional
+    fun saveNewUser(userNameCandidate: String, userRole: String): User {
+        val existedUser = userRepository.findByName(userNameCandidate)
+        val name = existedUser?.let {
+            val prefix = "${userNameCandidate}_"
```

Comment:
I don't insist, but may be this separator  `_` could be extracted to variable

Reply:
done